src/services/db/__tests__/GitFileCommitService.spec.ts

```diff
@@ -0,0 +1,555 @@
+          .spyOn(gitFileSystemService, "update")
+          .mockResolvedValue(okAsync("new-sha"))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        // Act
+        const result = await gitFileCommitService.update(sessionData, {
+          fileContent,
+          sha,
+          fileName,
+          directoryName,
+        })
+
+        // Assert
+        const filePath = directoryName
+          ? `${directoryName}/${fileName}`
+          : fileName
+        expect(updateSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          filePath,
+          fileContent,
+          sha,
+          sessionData.isomerUserId,
+          STAGING_BRANCH
+        )
+        expect(pushSpy).toHaveBeenCalledWith(sessionData, expect.any(Boolean))
+        expect(result).toEqual({ newSha: "new-sha" })
+      })
+
+      it("should update a file and commit only to staging branch when not whitelisted", async () => {
+        const fileContent = "updated file content"
+        const sha = "old-sha"
+        const fileName = "file.txt"
+        const directoryName = "directory"
+
+        const updateSpy = jest
+          .spyOn(gitFileSystemService, "update")
+          .mockResolvedValue(okAsync("new-sha"))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        jest
+          .spyOn(gbUtils, "isReduceBuildTimesWhitelistedRepo")
+          .mockReturnValue(false)
+
+        // Act
+        const result = await gitFileCommitService.update(sessionData, {
+          fileContent,
+          sha,
+          fileName,
+          directoryName,
+        })
+
+        // Assert
+        const filePath = directoryName
+          ? `${directoryName}/${fileName}`
+          : fileName
+        expect(updateSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          filePath,
+          fileContent,
+          sha,
+          sessionData.isomerUserId,
+          STAGING_BRANCH
+        )
+        expect(updateSpy).toHaveBeenCalledTimes(1) // update is called only once for the staging branch
+        expect(pushSpy).toHaveBeenCalledWith(sessionData, false) // push is called with shouldUpdateStagingLite as false
+        expect(result).toEqual({ newSha: "new-sha" })
+      })
+    })
+
+    describe("deleteDirectory", () => {
+      it("should delete a directory and push to GitHub", async () => {
+        // Arrange
+        const directoryName = "directory"
+
+        const deleteSpy = jest
+          .spyOn(gitFileSystemService, "delete")
+          .mockResolvedValue(okAsync(""))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        // Act
+        await gitFileCommitService.deleteDirectory(sessionData, {
+          directoryName,
+        })
+
+        // Assert
+        expect(deleteSpy).toHaveBeenCalledWith(
```

Comment:
should assert if delete was called twice one for each branch right?

Reply:
sames as above